src/replace_allocate.cpp

```diff
@@ -104,19 +104,17 @@ void replace_allocate::apply(module& m) const
             continue;
 
         auto s = ins->get_shape();
-
-        if(not main_offload_copy and model.needs_out_params() and contains(mod_output_names, ins))
+        if(not main_offload_copy and not(m.use_local_alloc) and model.needs_out_params() and
```

Comment:
Just take the compliment of the set of either outcome for allocs

not main_offload_copy and not(m.use_local_alloc) - > not(main_offload_copy or m.use_local_alloc)